contracts/lib/platform/EthGatedMarket.sol

```diff
@@ -44,23 +44,23 @@ contract EthGatedMarket is EthAdapter {
         gatewaySpecifier = specifier;
         setAdapterRate(adapterRate);
         bondAllow = true;
-        return gatewayToken;
+        return address(gatewayToken);
     }
 
-    function setMarket(TokenAdapter _market) onlyOwner {
+    function setMarket(TokenAdapter _market) public onlyOwner {
         marketFactory = _market;
     }
 
     // bond to obtain gateway tokens in exchange for eth, able to bond to gated curves
-    function gatewayBond(uint quantity) public payable {
+    function gatewayBond(uint quantity) public payable{
         require(bondAllow, "bond not allowed");
         super.bond(msg.sender, gatewaySpecifier, quantity);
     }
 
     // unbond to obtain eth in exchange for gateway tokens
-    function gatewayUnbond(uint quantity) public {
+    function gatewayUnbond(uint quantity) public payable{
         require(unbondAllow, "unbond not allowed");
-        super.unbond(msg.sender, gatewaySpecifier, quantity);
+        super.unbond1(msg.sender, gatewaySpecifier, quantity);
```

Comment:
suspected namespacing issues so renamed function to unbond1 to prevent other bond function to be called correctly.